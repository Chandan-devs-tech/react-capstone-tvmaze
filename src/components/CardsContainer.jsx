import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight } from '@fortawesome/free-solid-svg-icons';
import { fetchShow } from '../redux/show/showSlice';
import '../styles/CardsContainer.css';

function CardsContainer() {
  const dispatch = useDispatch();
  const allTvShow = useSelector((state) => state.show.shows);
  const [search, setSearch] = useState('');

  useEffect(() => {
    if (allTvShow.length === 0) {
      dispatch(fetchShow());
    }
  }, [dispatch, allTvShow.length]);

  return (
    <div className="mainContainer">
      <div className="searchBar">
        <form className="form">
          <input
            type="search"
            className="searchBox"
            placeholder="Search Shows and People"
            onChange={(e) => setSearch(e.target.value)}
          />
          <div className="small-4 columns">
            <button type="submit" className="medium button postfix" alt="Search">
              <i className="fa fa-search fa-lg" aria-hidden="true" />
            </button>
          </div>
        </form>
      </div>
      <div className="cardContainer">
        {allTvShow
          .filter((item) => search.trim() === ''
            || item.name.toLowerCase().startsWith(search.toLowerCase()))
          .map((show) => (
            <div className="cardShow" key={show.id}>
              <div className="cardImgContainer">
                <img
                  src={show.image}
                  alt="showImage"
                  className="cardImg"
                />
                <Link to={`/details/${show.id}`}>
                  <FontAwesomeIcon
                    icon={faCircleArrowRight}
                    className="nextIcon"
                  />
                </Link>
              </div>
              <div className="cardDetail">
                <p className="name">
                  Name:
                  {show.name}
                </p>
                <p className="rate">
                  Rate:
                  {show.rating}
                </p>
                <p className="time">
                  Time:
                  {show.time}
                </p>
              </div>
            </div>
          ))}
      </div>
    </div>
  );
}

export default CardsContainer;
