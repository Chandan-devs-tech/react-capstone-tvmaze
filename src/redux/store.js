import { configureStore } from '@reduxjs/toolkit';
import showsReducer from './show/showSlice';

const store = configureStore({
  reducer: {
    show: showsReducer,
  },
});

export default store;
