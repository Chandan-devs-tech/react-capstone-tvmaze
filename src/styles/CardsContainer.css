.mainContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.searchBar {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 15px;
  background-color: #3c948b;
}

.form {
  width: 40%;
}

.searchBox {
  width: 100%;
  padding: 10px;
  border-radius: 20px;
  border: transparent;
  text-align: center;
  font-size: 15px;
}

.cardContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  width: 100%;
}

.cardImg {
  width: 100%;
  height: 100%;
}

.cardImgContainer {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 325px;
  gap: 7px;
}

.cardShow {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 2px solid #3c948b;
  padding: 15px;
}

.cardDetail {
  color: #fff;
  padding-top: 10px;
}

.nextIcon {
  color: #fff;
  font-size: 25px;
}
