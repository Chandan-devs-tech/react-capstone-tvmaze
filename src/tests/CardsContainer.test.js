import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CardsContainer from '../components/CardsContainer';
import store from '../redux/store';
import '@testing-library/jest-dom';

describe('CardsContainer test', () => {
  const the100 = {
    name: 'The 100',
    id: 4,
  };
  it('It checks that the component is changed', () => {
    const myTree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CardsContainer
              key={the100.id}
              id={the100.id}
              name={the100.name}
            />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(myTree).toMatchSnapshot();
  });
});
