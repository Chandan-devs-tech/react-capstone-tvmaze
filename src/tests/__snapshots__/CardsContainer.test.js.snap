// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer test It checks that the component is changed 1`] = `
<div
  className="mainContainer"
>
  <div
    className="searchBar"
  >
    <form
      className="form"
    >
      <input
        className="searchBox"
        onChange={[Function]}
        placeholder="Search Shows and People"
        type="search"
      />
      <div
        className="small-4 columns"
      >
        <button
          alt="Search"
          className="medium button postfix"
          type="submit"
        >
          <i
            aria-hidden="true"
            className="fa fa-search fa-lg"
          />
        </button>
      </div>
    </form>
  </div>
  <div
    className="cardContainer"
  />
</div>
`;
